resend/actions/submitEmail.ts

```diff
@@ -0,0 +1,34 @@
+import { AppContext } from "../mod.ts";
+import { CreateEmailOptions, CreateEmailResponse } from "../utils/types.ts";
+
+export interface Props {
+  payload: CreateEmailOptions;
```

Comment:
isn't payload too generic for props? why not passing prop by prop flattening the payload?